framework/test/unit/state_machine/state_machine.spec.ts

```diff
@@ -205,6 +205,49 @@ describe('state_machine', () => {
 			expect(events).toHaveLength(1);
 			expect(events[0].toObject().topics[0]).toEqual(transaction.id);
 		});
+
+		it('should rollback state if afterCommandExecute fails', () => {
+			const events = [
+				{
+					moduleID: utils.intToBuffer(3, 4),
+					typeID: Buffer.from([0, 0, 0, 0]),
+					data: utils.getRandomBytes(20),
+					topics: [utils.getRandomBytes(32), utils.getRandomBytes(20)],
+				},
+				{
+					moduleID: utils.intToBuffer(4, 4),
+					typeID: Buffer.from([0, 0, 0, 0]),
+					data: utils.getRandomBytes(20),
+					topics: [utils.getRandomBytes(32), utils.getRandomBytes(20)],
+				},
+				{
+					moduleID: utils.intToBuffer(2, 4),
+					typeID: Buffer.from([0, 0, 0, 0]),
+					data: utils.getRandomBytes(20),
+					topics: [utils.getRandomBytes(32)],
+				},
+			];
+
+			events.map(e => eventQueue.add(e.moduleID, e.typeID, e.data, e.topics));
+
+			const snapshotID = eventQueue.createSnapshot();
+
+			eventQueue.add(
+				utils.intToBuffer(3, 4),
+				Buffer.from([0, 0, 0, 1]),
+				utils.getRandomBytes(100),
+				[utils.getRandomBytes(32)],
+			);
+
+			mod.afterCommandExecute.mockImplementation(() => {
+				throw new Error('error');
+			});
+
```

Comment:
i think it should call `await stateMachine.executeTransaction(ctx);` to see if `mod.afterCommandExecute` fails, it rolls back to the original state/events

Reply:
Updated the test.